src/transaction/guards.ts

```diff
@@ -32,12 +36,33 @@ export const isTxInput = (
   return Buffer.isBuffer(txHash) && Number.isInteger(outputIndex)
 }
 
+// export const isAsset = (
+//   test: any,
+// ): test is Asset => test instanceof Map
+//   && Object.entries(test).every(([assetName, amount]) => Buffer.isBuffer(assetName) && isBigNumber(amount))
+
+// export const isMultiAsset = (
+//   test: any,
+// ): test is MultiAsset => test instanceof Map
+//   && Object.entries(test).every(([policyId, asset]) => Buffer.isBuffer(policyId) && isAsset(asset))
+
 export const isTxOutput = (
```

Comment:
I initially wanted to write guards for whole `TxOutput`, but making a whole structure check and returning `true/false` seems to me like we are throwing away a lot of information that can be used to give more specific errors.

Check `parseTxOutputs` function, which then calls `parseMultiAsset` and `parseAsset`. Structure is checked during iteration and when error occurs, we can provide more descriptive error.

@PeterBenc  what do you think?

Reply:
Agreed, we could simplify the `isTxOutput` guard to just validate if the output is an array and has two parts of which the first one (address) is a buffer and the second one is either coin or object.`[address, coinsOrMa]`, if not throw error. and dont validate the supparts. In the next step we could check the `coins` or `asset` one by one.